src/protocol/ipa/mod.rs

```diff
@@ -714,6 +806,115 @@ pub mod tests {
         assert_eq!(MAX_BREAKDOWN_KEY, result.len() as u128);
     }
 
+    struct TestRawDataRecord {
```

Comment:
There is a lot of code duplication here.

Reply:
Yes, I am in the process of eliminating the `random_ipa_no_result_check` test, and merging it into this one which actually does test outputs.

Reply:
Here's the PR: https://github.com/private-attribution/ipa/pull/480

...but in the process of merging I discovered we have a logic error... and the test fails for per_user_cap > 1... so this is just step one to get this test up so that we can figure out where the issue is.